--- libnfsidmap-0.23/idmapd.conf.orig	2009-07-29 22:19:06.000000000 +0200
+++ libnfsidmap-0.23/idmapd.conf	2010-08-12 20:46:04.110823295 +0200
@@ -15,10 +15,12 @@
 # must be included in the list!
 #Local-Realms = 
 
+#Cache-Expiration = 600
+  
 [Mapping]
 
-#Nobody-User = nobody
-#Nobody-Group = nobody
+Nobody-User = nobody
+Nobody-Group = nobody
 
 [Translation]
 
@@ -28,7 +30,7 @@
 # is a dynamically loadable plugin library.
 # New methods may be defined and inserted in the list.
 # The default is "nsswitch".
-#Method = nsswitch
+Method = nsswitch
 
 # Optional.  This is a comma-separated, ordered list of
 # translation methods to be used for translating GSS
@@ -52,13 +54,13 @@
 # The following are used only for the "umich_ldap" Translation Method.
 #-------------------------------------------------------------------#
 
-[UMICH_SCHEMA]
+#[UMICH_SCHEMA]
 
 # server information (REQUIRED)
-LDAP_server = ldap-server.local.domain.edu
+#LDAP_server = ldap-server.local.domain.edu
 
 # the default search base (REQUIRED)
-LDAP_base = dc=local,dc=domain,dc=edu
+#LDAP_base = dc=local,dc=domain,dc=edu
 
 #-----------------------------------------------------------#
 # The remaining options have defaults (as shown)
